components/link/link.js

```diff
@@ -139,13 +157,27 @@ class Link extends FocusMixin(LitElement) {
 			'truncate': this.lines > 0
 		};
 		const styles = (this.lines > 0) ? { '-webkit-line-clamp': this.lines } : {};
-		return html`<a
+		const target = this.newWindow && this.target === undefined
+			? '_blank'
+			: this.target;
+
+		return html`
+			<a
 				aria-label="${ifDefined(this.ariaLabel)}"
 				class="${classMap(linkClasses)}"
 				style="${styleMap(styles)}"
 				?download="${this.download}"
 				href="${ifDefined(this.href)}"
-				target="${ifDefined(this.target)}"><slot></slot></a>`;
+				target="${ifDefined(target)}"
+			>
+				<slot></slot>
+				${target === '_blank' ? html`
+					<span class="d2l-offscreen">${this.localize('components.link.open-in-new-window')}</span>
+				` : nothing}
+				${this.newWindow ? html`
+					<d2l-icon class="d2l-new-window" icon="tier1:new-window"></d2l-icon>
+				` : nothing}
```

Comment:
This will have to be something like this (or some better less annoying method I don't know about), so that the icon can't wrap to the next line by itself.
```suggestion
				<span style="white-space: nowrap;">
					<span style="white-space: normal;"><slot></slot></span>${this.newWindow ? html`&nbsp;<d2l-icon class="d2l-new-window" icon="tier1:new-window"></d2l-icon>
					` : nothing}
					${target === '_blank' ? html`
						<span class="d2l-offscreen">${this.localize('components.link.open-in-new-window')}</span>
					` : nothing}
				</span>
```
So we get this:
<img width="198" alt="Screenshot 2023-09-13 at 3 25 54 PM" src="https://github.com/BrightspaceUI/core/assets/1289042/4230e834-db1e-428c-93cc-2ee8b0a3257d">

Instead of this:
<img width="204" alt="Screenshot 2023-09-13 at 3 26 17 PM" src="https://github.com/BrightspaceUI/core/assets/1289042/7edcca0b-a202-47e2-8c62-4e6166bda7d7">

Reply:
Oooh, good call. @NicholasHallman can you add a vdiff that covers that case to make sure we don't accidentally break it in the future?

Reply:
Actually, I think this entire thing has to be one line, otherwise any trailing implicit space is included in the link.

<img width="155" alt="Screenshot 2023-09-13 at 3 44 12 PM" src="https://github.com/BrightspaceUI/core/assets/1289042/240da254-925b-471e-a89a-c927d0213c73">

vs

<img width="162" alt="Screenshot 2023-09-13 at 3 43 57 PM" src="https://github.com/BrightspaceUI/core/assets/1289042/e26dfef3-97e5-49af-acc4-96c4864b1589">

Reply:
Your solution works and I couldn't find anything simpler so I've commited it, thanks!

Reply:
> Actually, I think this entire thing has to be one line, otherwise any trailing implicit space is included in the link.

The only way I've found to fix this is to add `display: inline-block` to the `<a>` tag. Any other solution for this?